### PR TITLE
Update docker deployment documentation

### LINKED ISSUE #1801: Update "How to run FerretDB on Docker" docs
### What should be done?

I believe that on the current version of FerretDB the environment doesn't work correctly, so it's need to be checked, and changed in a docs.

## DoD:
- Check if everything works correctly after doing everything step by step from documentation.
- If it doesn't work - find a cause and update the docs.

### PR DESCRIPTION
**Pull request by noisersup**
# Description

Closes #1801.

## Readiness checklist

* [ ] I added unit tests for new functionality or bug fixes.
* [ ] I added integration tests for new functionality or bug fixes.
* [ ] I added compatibility tests for new functionality or bug fixes.
* [ ] I made spot refactorings.
* [x] I updated user documentation.
* [x] I ran `task all`, and it passed.
* [ ] I added/updated comments for both exported and unexported top-level declarations (functions, types, etc).
* [ ] I checked comments rendering with `task godocs`.
* [x] I ensured that the title is good enough for the changelog.
* [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Assignee, Labels, Project and project's Sprint fields.
* [x] I marked all done items in this checklist.

### CODE REVIEW

**Comment by noisersup** [comment]
So, the first issue is that I forgot to pull the latest version so I had some issues with pg hostname:
```MongoServerError: failed to connect to `host=hostname user=username database=ferretdb`: hostname resolving error (lookup hostname: no such host)```

This is fixed after fetching the new image so I updated file to remind of using `docker compose pull` command.

The other problem is that we don't mention neither in docs nor README file how to authenicate. We instruct to provide  `POSTGRES_USER` and `POSTGRES_PASSWORD`, but afterwards we tell users to run mongosh without any addiitonal flags.

I will change the mongosh command to use `-u` and `-p` flags, and write a side note that FerretDB passes authentication details to the backend database, and mention https://docs.ferretdb.io/security/#authentication for more details.

Please let me know if we prefer to show `postgres://username:password@postgres:5432/ferretdb` as the default way of authentication

File: README.md [hunk 1]
```diff
@@ -34,7 +34,9 @@ and [contributing guidelines](CONTRIBUTING.md).
 ## Quickstart
 
 These steps describe a quick local setup.
-They are not suitable for most production use-cases because they keep all data inside containers.
+They are not suitable for most production use-cases because they keep all data
+inside containers and don't [encrypt incoming connections](https://docs.ferretdb.io/security/#securing-connections-with-tls/).
```
Comment (AlekSi) [inline README.md:38]: Please use `../security.md`. Never include a domain name

**Review by AlekSi — CHANGES_REQUESTED** [review]

File: README.md (hunk 1 above)
Comment (noisersup) [reply README.md:38]: That's the README file, accessible from github view

Comment (AlekSi) [reply README.md:38]: Oops! My bad, sorry

File: website/docs/quickstart_guide/docker.md [hunk 2]
```diff
@@ -41,13 +43,17 @@ They are not suitable for most production use-cases because they keep all data i
...
 3. If you have `mongosh` installed, just run it to connect to FerretDB.
    If not, run the following command to run `mongosh` inside the temporary MongoDB container, attaching to the same Docker network:
 
    ```sh
-   docker run --rm -it --network=ferretdb --entrypoint=mongosh mongo mongodb://ferretdb/
+   docker run --rm -it --network=ferretdb --entrypoint=mongosh mongo mongodb://ferretdb/ -u username -p password
    ```
 
-You can also install with FerretDB with the `.deb` and `.rpm` packages
+   FerretDB will use credentials passed in mongosh flags or mongodb URI to authenticate to the PostgreSQL database.
```
Comment (AlekSi) [inline website/docs/quickstart_guide/docker.md:55]:
```suggestion
   FerretDB will use credentials passed in `mongosh` flags or MongoDB URI to authenticate to the PostgreSQL database.
```

**Review by AlekSi — DISMISSED** [review]

File: README.md [hunk 3]
```diff
@@ -70,12 +72,13 @@ They are not suitable for most production use-cases because they keep all data i
...
 
 2. Start services with `docker compose up -d`.
+   To be sure that you've fetched the latest version you can precede the command with `docker compose pull`.
 
 3. If you have `mongosh` installed, just run it to connect to FerretDB.
    If not, run the following command to run `mongosh` inside the temporary MongoDB container, attaching to the same Docker network:
 
    ```sh
-   docker run --rm -it --network=ferretdb --entrypoint=mongosh mongo mongodb://ferretdb/
+   docker run --rm -it --network=ferretdb --entrypoint=mongosh mongo mongodb://ferretdb/ -u username -p password
```
Comment (w84thesun) [inline README.md:81]: I ran setup step by step until here I get `MongoServerError: Unsupported mechanism 'PLAIN'` error. Does it work for you?

**Review by w84thesun — COMMENTED** [review]
Asked about mongo shell connection problem.

File: README.md (hunk 3 above)
Comment (noisersup) [reply README.md:81]: Nice catch! 

I was sure that I've connected with success, but probably I've changed too many things and didn't track the changes. 

Adding the `--authenticationMechanism=PLAIN` flag fix the issue. To ensure that everything works correctly I've run everything on seperate environment :smile:

File: website/docs/quickstart_guide/docker.md [hunk 4]
```diff
@@ -41,13 +43,17 @@ They are not suitable for most production use-cases because they keep all data i
    `ferretdb` runs FerretDB.
 
 2. Start services with `docker compose up -d`.
+   To be sure that you've fetched the latest version you can precede the command with `docker compose pull`.
 
 3. If you have `mongosh` installed, just run it to connect to FerretDB.
```
Comment (rumyantseva) [inline website/docs/quickstart_guide/docker.md:48]: "just run it to connect to FerretDB" doesn't seem enough anymore as we need to mention username, password and authentication method.

**Review by rumyantseva — COMMENTED** [review]
👍👍👍 I tried this, and it works, I'd rephrase "just run it to connect to FerretDB" as it's not that "just" anymore :)

File: README.md [hunk 5]
```diff
@@ -70,12 +72,13 @@ They are not suitable for most production use-cases because they keep all data i
    `ferretdb` runs FerretDB.
 
 2. Start services with `docker compose up -d`.
+   To be sure that you've fetched the latest version you can precede the command with `docker compose pull`.
```
Comment (chilagrow) [inline README.md:75]:
```suggestion
   Fetch the latest version with the command `docker compose pull`.
```
Just less words.

File: website/docs/quickstart_guide/docker.md [hunk 6]
```diff
@@ -41,13 +43,17 @@ They are not suitable for most production use-cases because they keep all data i
    `ferretdb` runs FerretDB.
 
 2. Start services with `docker compose up -d`.
+   To be sure that you've fetched the latest version you can precede the command with `docker compose pull`.
```
Comment (chilagrow) [inline website/docs/quickstart_guide/docker.md:46]:
```suggestion
   Fetch the latest version with the command `docker compose pull`.
```

File: website/docs/quickstart_guide/docker.md [hunk 7]
```diff
@@ -41,13 +43,17 @@ They are not suitable for most production use-cases because they keep all data i
...
    If not, run the following command to run `mongosh` inside the temporary MongoDB container, attaching to the same Docker network:
 
    ```sh
-   docker run --rm -it --network=ferretdb --entrypoint=mongosh mongo mongodb://ferretdb/
+   docker run --rm -it --network=ferretdb --entrypoint=mongosh mongo mongodb://ferretdb/ -u username -p password --authenticationMechanism=PLAIN
    ```
 
-You can also install with FerretDB with the `.deb` and `.rpm` packages
+   FerretDB will use credentials passed in `mongosh` flags or MongoDB URI to authenticate to the PostgreSQL database.
+   See [Security#Authentication](/security.md#authentication) for more details.
```
Comment (chilagrow) [inline website/docs/quickstart_guide/docker.md:56]: Did you check this link works? :hugs:

**Review by chilagrow — COMMENTED** [review]
Some minor comments :hugs: Thanks for the doc update!

File: README.md [hunk 8]
```diff
@@ -70,12 +72,13 @@ They are not suitable for most production use-cases because they keep all data i
...
 
 3. If you have `mongosh` installed, just run it to connect to FerretDB.
    If not, run the following command to run `mongosh` inside the temporary MongoDB container, attaching to the same Docker network:
 
    ```sh
-   docker run --rm -it --network=ferretdb --entrypoint=mongosh mongo mongodb://ferretdb/
+   docker run --rm -it --network=ferretdb --entrypoint=mongosh mongo mongodb://ferretdb/ -u username -p password --authenticationMechanism=PLAIN
```
Comment (AlekSi) [inline README.md:81]: I think we should use MongoDB URI everywhere we can: `mongodb://username:password@ferretdb/?authMechanism=PLAIN`

File: website/docs/quickstart_guide/docker.md (hunk 7 above)
Comment (AlekSi) [reply website/docs/quickstart_guide/docker.md:56]: I'm writing a bit more about it ATM, but let's use relative file paths as they seem to work everywhere, including editors/IDEs (Ctrl+click works) and Docusaurus. In that case, it should be `../security.md#authentication`.

Comment (noisersup) [reply website/docs/quickstart_guide/docker.md:56]: > Did you check this link works?

Yes

> let's use relative file paths as they seem to work everywhere, including editors/IDEs (Ctrl+click works) and Docusaurus

Good point, I'll change it then

File: README.md (hunk 5 above)
Comment (noisersup) [reply README.md:75]: In that case I believe we should put `docker compose pull` before `docker compose up`, as it needs to be ran first. The other thing is that it's optional to run `docker compose pull`

File: README.md [hunk 9]
```diff
@@ -70,12 +72,20 @@ They are not suitable for most production use-cases because they keep all data i
    `ferretdb` runs FerretDB.
 
 2. Start services with `docker compose up -d`.
+   To be sure that you've fetched the latest version you can precede the command with `docker compose pull`.
 
-3. If you have `mongosh` installed, just run it to connect to FerretDB.
-   If not, run the following command to run `mongosh` inside the temporary MongoDB container, attaching to the same Docker network:
+3. If you have `mongosh` installed, just run it to connect to FerretDB. It will use credentials passed in `mongosh` flags or MongoDB URI 
+   to authenticate to the PostgreSQL database. You'll also need to set `authMechanism` to `PLAIN`. 
+   The example uri would look like:
```
Comment (Fashander) [inline README.md:79]:
```suggestion
   The example URI would look like: 
```

File: website/docs/quickstart_guide/docker.md [hunk 10]
```diff
@@ -41,13 +43,25 @@ They are not suitable for most production use-cases because they keep all data i
    `ferretdb` runs FerretDB.
 
 2. Start services with `docker compose up -d`.
+   To be sure that you've fetched the latest version you can precede the command with `docker compose pull`.
 
-3. If you have `mongosh` installed, just run it to connect to FerretDB.
-   If not, run the following command to run `mongosh` inside the temporary MongoDB container, attaching to the same Docker network:
+3. If you have `mongosh` installed, just run it to connect to FerretDB. It will use credentials passed in `mongosh` flags or MongoDB URI 
+   to authenticate to the PostgreSQL database. You'll also need to set `authMechanism` to `PLAIN`. 
+   The example uri would look like:
```
Comment (Fashander) [inline website/docs/quickstart_guide/docker.md:50]:
```suggestion
   The example URI would look like: 
```

Same here too

**Review by Fashander — APPROVED** [review]

**Review by w84thesun — DISMISSED** [review]

File: README.md [hunk 11]
```diff
@@ -70,12 +72,23 @@ They are not suitable for most production use-cases because they keep all data i
...
 2. Start services with `docker compose up -d`.
+   To be sure that you've fetched the latest version you can precede the command with `docker compose pull`.
 
 3. If you have `mongosh` installed, just run it to connect to FerretDB.
-   If not, run the following command to run `mongosh` inside the temporary MongoDB container, attaching to the same Docker network:
+   It will use credentials passed in `mongosh` flags or MongoDB URI to authenticate to the PostgreSQL database.
+   You'll also need to set `authMechanism` to `PLAIN`.
+   The example URI would look like:
+
+   ``` sh
```
Comment (AlekSi) [inline README.md:82]:
```suggestion
   ```sh
```

Otherwise, it will not render correctly

File: README.md [hunk 12]
```diff
@@ -70,12 +72,23 @@ They are not suitable for most production use-cases because they keep all data i
...
+   mongodb://username:password@localhost/ferretdb?authMechanism=PLAIN"`
```
Comment (AlekSi) [inline README.md:83]:
```suggestion
   mongodb://username:password@localhost/ferretdb?authMechanism=PLAIN
```

File: website/docs/quickstart_guide/docker.md [hunk 13]
```diff
@@ -41,13 +43,25 @@ They are not suitable for most production use-cases because they keep all data i
...
+   To be sure that you've fetched the latest version you can precede the command with `docker compose pull`.
 
 3. If you have `mongosh` installed, just run it to connect to FerretDB.
-   If not, run the following command to run `mongosh` inside the temporary MongoDB container, attaching to the same Docker network:
+   It will use credentials passed in `mongosh` flags or MongoDB URI
+   to authenticate to the PostgreSQL database.
+   You'll also need to set `authMechanism` to `PLAIN`.
+   The example URI would look like:
+
+   ``` sh
```
Comment (AlekSi) [inline website/docs/quickstart_guide/docker.md:54]:
```suggestion
   ```sh
```

File: website/docs/quickstart_guide/docker.md [hunk 14]
```diff
@@ -41,13 +43,25 @@ They are not suitable for most production use-cases because they keep all data i
...
+   mongodb://username:password@localhost/ferretdb?authMechanism=PLAIN"`
```
Comment (AlekSi) [inline website/docs/quickstart_guide/docker.md:55]:
```suggestion
   mongodb://username:password@localhost/ferretdb?authMechanism=PLAIN
```

**Review by Fashander — APPROVED** [review]

**Review by w84thesun — APPROVED** [review]

File: README.md (hunk 5 above)
Comment (chilagrow) [reply README.md:75]: Yeah we can do that :+1:

File: website/docs/quickstart_guide/docker.md [hunk 15]
```diff
@@ -41,13 +43,25 @@ They are not suitable for most production use-cases because they keep all data i
...
 
 3. If you have `mongosh` installed, just run it to connect to FerretDB.
-   If not, run the following command to run `mongosh` inside the temporary MongoDB container, attaching to the same Docker network:
+   It will use credentials passed in `mongosh` flags or MongoDB URI
+   to authenticate to the PostgreSQL database.
+   You'll also need to set `authMechanism` to `PLAIN`.
+   The example URI would look like:
 
    ```sh
-   docker run --rm -it --network=ferretdb --entrypoint=mongosh mongo mongodb://ferretdb/
```
Comment (chilagrow) [inline website/docs/quickstart_guide/docker.md:49]: Let's remove `sh` in \`\`\`sh. It's not a bash :hugs:

Comment (AlekSi) [reply website/docs/quickstart_guide/docker.md:49]: Well, that's a shell command with an URL as an argument.

@noisersup CI fails now

Comment (noisersup) [reply website/docs/quickstart_guide/docker.md:49]: It's not a shell command, but github view makes it to look like one :)

I've sent the commit to quickly and didn't cover everything yet. It will be ready in a moment

**Review by rumyantseva — APPROVED** [review]
LGTM!

Maybe @Fashander could check/help if we need more clarity for users

**Review by w84thesun — APPROVED** [review]
:shipit:

**Review by Fashander — APPROVED** [review]

**Review by AlekSi — APPROVED** [review]
